app/components/ClientTaskList/ClientTaskList.tsx

```diff
@@ -14,21 +26,55 @@ import { TaskItem } from './TaskItem';
  * @returns {React.ReactElement} A JSX element representing the list of tasks.
  */
 const ClientTaskList = (): React.ReactElement => {
-  const { tasks, fetchTasks } = useTaskStore();
+  const { tasks, fetchTasks, reorderTask } = useTaskStore();
+
+  // https://docs.dndkit.com/api-documentation/sensors
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 8,
+      },
+    })
+  );
 
   // TODO: fetch tasks differently - not in a useEffect
   useEffect(() => {
     fetchTasks();
   }, [fetchTasks]);
 
+  // dnd setup
+  const handleDragEnd = async (event: DragEndEvent) => {
+    const { active, over } = event;
+
+    if (!over) return;
+
+    if (active.id !== over.id) {
+      reorderTask(String(active.id), String(over.id));
```

Comment:
I like keeping the bulk of the logic in the store - especially because these changes will be reflected in the DB